enterprise/api/v1/views.py

```diff
@@ -213,6 +213,79 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course.
```

Comment:
I would doc here some flavor of the non transctional-ness of this endpoint so we know what to expect when sending a request here. e.g, if some of the enrollments fail, we will still have enrolled a partial set of users, correct? stuff like that.

Reply:
also if it does fail to enroll some of the learners, will the caller be expected to know which learner+course enrollments failed, or just which learners? in the least I think a requirement is to send back the set of learners that failed to enroll, to the MFE. will the license manager endpoint handle that vs this endpoint?

Reply:
A failure at this point is bad because learners are expected to be pre-vetted by License manager in terms of their eligibility to enroll in the desired courses. What are we supposed to do if an enrollment is successfully created, but a licensed enrollment isn't? Beyond experiencing an exception that bubbles up and completely crashes the request, we return the number of users enrolled in the number of courses, which would give the frontend an idea of what happened.

Reply:
yeah, I think eventually we have to be transactionally safe in this endpoint, but for this PR I am not too worried. If we do fail to create licesnsed enrollment objects, I wonder if re enrolling them at a later time again will work, will it? if so we are not too worried yet I would say

however if an attempt to enrol the same learner into the same course again fails, then we would get into a bad state

Reply:
Ok, so is there a test for this case? The test case being a failure that succeeds later?